torch/utils/data/datapipes/dataframe/dataframe_wrapper.py

```diff
@@ -1,53 +1,64 @@
-try:
-    import pandas  # type: ignore[import]
-
-    # pandas used only for prototyping, will be shortly replaced with TorchArrow
-    WITH_PANDAS = True
-except ImportError:
-    WITH_PANDAS = False
-
+_pandas = None
+_WITH_PANDAS = None
+
+def _try_import_pandas() -> bool:
+    try:
+        import pandas  # type: ignore[import]
+        global _pandas
+        _pandas = pandas
+        return True
+    except ImportError:
+        return False
+
+
+# pandas used only for prototyping, will be shortly replaced with TorchArrow
+def _with_pandas() -> bool:
```

Comment:
Drive by comment: couldn't the None-ness of `_pandas` be the flag?
```
_pandas = None
def _get_pandas():
  global _pandas
  if _pandas is None:
    try:
      import pandas  # type: ignore[import]
      _pandas = pandas
    except ImportError:
      raise Exception("DataFrames prototype requires pandas to function")

  return _pandas

%timeit _get_pandas()
```

And then you could just call `_get_pandas().foo(...)`

Reply:
Good idea, though my initial concern were that it will try to re-import pandas every time before raising an exception, but I guess it's not at all on a performance critical path

Reply:
Although, let me land it as is and will do a refactor in follow up